src/pages/ProfilesExplorerView/infrastructure/withPreventInvalidQuery.ts

```diff
@@ -0,0 +1,58 @@
+import { LoadingState } from '@grafana/data';
+import { sceneGraph, SceneQueryRunner } from '@grafana/scenes';
+import { parseQuery } from '@shared/domain/url-params/parseQuery';
+import { logger } from '@shared/infrastructure/tracking/logger';
+
+export function withPreventInvalidQuery(queryRunner: SceneQueryRunner) {
+  queryRunner.addActivationHandler(() => {
+    const { profileTypeId, labelSelector } = queryRunner.state.queries[0];
+
+    if (!profileTypeId) {
+      queryRunner.setState({
+        queries: [{ refId: 'null' }],
+        data: buildErrorData(queryRunner, 'Missing profile type!'),
```

Comment:
`[{ refId: 'null' }]` prevents the query to be executed but, unfortunately, even if we build some synthetic "errored" data, the UI doesn't display the error message.